hphp/runtime/ext/escape_hatch/ext_escape_hatch.cpp

```diff
@@ -0,0 +1,500 @@
+#include <folly/Chrono.h>
+#include <folly/io/async/AsyncSocket.h>
+#include <folly/io/IOBuf.h>
+#include <folly/io/IOBufQueue.h>
+
+#include "hphp/runtime/ext/asio/asio-external-thread-event.h"
+#include "hphp/runtime/ext/asio/socket-event.h"
+#include "hphp/runtime/ext/extension.h"
+
+#include "hphp/util/compatibility.h"
+#include "hphp/util/logger.h"
+
+using std::shared_ptr;
+using folly::AsyncSocket;
+using folly::AsyncTimeout;
+using folly::EventBase;
+using folly::IOBuf;
+using folly::IOBufQueue;
+using folly::SocketAddress;
+using folly::stringPrintf;
+
+#define EH_HEADER_SIZE 8
+#define EH_MAX_MSG_SIZE 0xFFFFFFFF
+#define LOGGER DLOG(INFO)
+#define EH_THREAD_CHECKS 0
+
+namespace HPHP {
+
+/**
+ * Escape Hatch: A simple asynchronous IPC mechanism intended for use with a
+ * sidecar process. Supports request+response or one-way requests over pooled
+ * [unix domain] sockets. Every message in either direction is framed with
+ * an 8 byte header:
+ * - Magic bytes 'E' then 'H'
+ * - A single byte; 0x00 for oneway, or 0x01 for response requested.
+ * - Another byte, ignored.
+ * - An unsigned 32 bit big endian integer indicating the length of the
+ *   payload to come.
+ */
+
+static const auto optPool = String::FromCStr("pool");
```

Comment:
the usual pattern is `const StaticString s_optPool("pool")`

`static` /could/ be added there, but we generally wrap the file-local stuff in `namespace {` instead

Reply:
done